screens/CameraRecording.tsx

```diff
@@ -2,48 +2,65 @@ import { Camera } from 'expo-camera'
+        }
+    }, [time, setTime, recording])
     if (!permissions) {
         return (<Text>Please give permissions</Text>)
     }
+    let dateTime = new Date(60 * 5 * 1000 - time);
+    const toTens = (num: number) => num >= 10 ? '' + num : '0' + num;
+    if (dateTime.getTime() <= 0 && recording) {
+        setRecording(false);
+        camera?.stopRecording();
+    }
     return (
         <View style={Styles.container}>
             <Camera style={Styles.camera}
-                type={cameraType}
-                onFacesDetected={(face) => setSeeingFace(face.faces.length > 0)}
-                ref={(camera) => setCamera(camera)}/>
-            <View style={Styles.bottomRow}>
-                <Button style={Styles.flip} onPress={() => {
-                    setCameraType(cameraType == Camera.Constants.Type.front ? Camera.Constants.Type.back : Camera.Constants.Type.front)
-                }}> 
-                    <Text style={Styles.text}>Flip</Text> 
-                </Button>
-                <Button style={Styles.takePicture} onPress={() => {
-                    if (camera) {
-                        camera.takePictureAsync()
-                            .then(pic => {
-                                console.log(pic.uri);
+            type={Camera.Constants.Type.front}
+            ref={(camera) => setCamera(camera)}
+            flashMode={Camera.Constants.FlashMode.auto}/>
+                <View style={Styles.bottomRow}>
+                    <Ionicons.Button style={Styles.icon} color='black' name={recording ? 'stop-circle' : "camera"} onPress={() => {
+                        if (camera && !recording) {
+                            setRecording(true);
+                            camera.recordAsync({ 
+                                maxFileSize: 150000000, 
+                                quality: Camera.Constants.VideoQuality['480p']
+                            }).then(vid => {
+                                console.log(vid.uri);
+                                setRecording(false);
                             })
-                    }
-                }}>
-                    <Text style={Styles.text}>Take Picture</Text>
-                </Button>
-                {!seeingFace && <Text style={Styles.faceShow}>Show your face</Text>}
-            </View>
-            
+                        }
+                        else if (camera && recording) {
+                            camera.stopRecording();
+                            setRecording(false);
+                        }
+                    }}>
+                        {recording ? 'Stop Recording' : 'Start Recording'}
+                    </Ionicons.Button>
+                </View>
+                {recording && <View style={Styles.totalTime}>
+                    <Text style={Styles.text}>
+                        Time Left: {toTens(dateTime.getMinutes()) + ':' + toTens(dateTime.getSeconds())}
+                    </Text>
+                </View> }
```

Comment:
Not a big thing but it might be nice to run a formatter/prettier on the code since the code is a little more awkwardly formatted

Reply:
Will do in follow-up pr!